bag_verifier/src/test/scala/uk/ac/wellcome/platform/archive/bagverifier/services/BagVerifierTestCases.scala

```diff
@@ -663,3 +686,58 @@ trait BagVerifierTestCases[BagLocation <: Location, BagPrefix <: Prefix[
       assertion(failedResult, summary)
     }
 }
+
+trait ReplicatedBagVerifierTestCases[
+  BagLocation <: Location,
+  BagPrefix <: Prefix[BagLocation],
+  Namespace
+] extends BagVerifierTestCases[
+      ReplicatedBagVerifier[BagLocation, BagPrefix],
+      ReplicatedBagVerifyContext[BagLocation, BagPrefix],
+      BagLocation,
+      BagPrefix,
+      Namespace
+    ] {
+  override def createBagRoot(
+    bagRoot: BagPrefix,
+    scrBagRoot: Option[BagPrefix]
+  ): ReplicatedBagVerifyContext[BagLocation, BagPrefix] =
+    ReplicatedBagVerifyContext(bagRoot, scrBagRoot.getOrElse(bagRoot))
+
+  it("fails a bag if it doesn't match original tag manifest") {
```

Comment:
Other test cases that would be useful:

* If the original and replicated `tagmanifest-sha256.txt` match, the bag is verified successfully. (Previously this was tested implicitly in the replicator – we'd know it worked if the other replicator tests passed.)

* What if we can't find the source bag at all?

Reply:
yep, I'll add them

Reply:
So, the first test is kinda done in that the `ReplicatedBagTestCases` extends `BagVerifierTestCases` and there first test is the passing one where they match. I'm saying _kinda_ done because the way it's done is the path of the bag and the path of the source are the same so of course the tagmanifest match